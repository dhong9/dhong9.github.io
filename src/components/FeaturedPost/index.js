import Typography from "@mui/material/Typography";
import Grid from "@mui/material/Grid";
import Card from "@mui/material/Card";
import CardActionArea from "@mui/material/CardActionArea";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";

// prop-types is a library for typechecking of props
import PropTypes from "prop-types";

export default function FeaturedPost(props) {
  const { post } = props;

  return (
    <Grid item xs={12} md={6}>
      <CardActionArea component="a" href="#">
        <Card sx={{ display: "flex" }}>
          <CardContent sx={{ flex: 1 }}>
            <Typography component="h2" variant="h5">
              {post.postName}
            </Typography>
            <Typography variant="subtitle1" color="text.secondary">
              {post.date}
            </Typography>
            <Typography variant="subtitle1" paragraph>
              {post.body}
            </Typography>
            <Typography variant="subtitle1" color="primary">
              Continue reading...
            </Typography>
          </CardContent>
          <CardMedia
            component="img"
            sx={{ width: 160, display: { xs: "none", sm: "block" } }}
            image={post.image}
            alt={post.imageText}
          />
        </Card>
      </CardActionArea>
    </Grid>
  );
}

// Typechecking props of FeaturedPost
FeaturedPost.propTypes = {
  post: PropTypes.shape({
    date: PropTypes.string,
    body: PropTypes.string,
    image: PropTypes.string,
    imageText: PropTypes.string,
    postName: PropTypes.string,
  }),
};

FeaturedPost.defaultProps = {
  post: {
    date: "",
    body: "",
    image: "https://source.unsplash.com/random",
    imageText: "main image description",
    postName: "",
  },
};
