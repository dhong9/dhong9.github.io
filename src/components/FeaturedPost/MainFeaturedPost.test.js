// React testing libraries
import renderer from "react-test-renderer";

// Component to test
import FeaturedPost from "components/FeaturedPost";

describe("FeaturedPost", () => {
  it("renders", () => {
    const posts = [{
        postName: "Bicycles",
        body: "It's healthy to bike to work everyday."
    }];
    const component = renderer.create(<FeaturedPost posts={posts} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
