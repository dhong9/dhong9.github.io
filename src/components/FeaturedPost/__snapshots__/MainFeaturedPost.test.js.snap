// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedPost renders 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-13i4rnv-MuiGrid-root"
>
  <a
    className="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
    href="#"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1oqv2c0-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-97bign-MuiCardContent-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        />
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 css-ecpxie-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-paragraph css-11j0rze-MuiTypography-root"
        />
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 css-x4wfr5-MuiTypography-root"
        >
          Continue reading...
        </h6>
      </div>
      <img
        alt="main image description"
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-fnq5ct-MuiCardMedia-root"
        src="https://source.unsplash.com/random"
      />
    </div>
    <span
      className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
    />
  </a>
</div>
`;
